hash_practice/exercises.py

```diff
@@ -1,19 +1,62 @@
+from collections import defaultdict 
+from heapq import nlargest 
 
 def grouped_anagrams(strings):
     """ This method will return an array of arrays.
         Each subarray will have strings which are anagrams of each other
-        Time Complexity: ?
-        Space Complexity: ?
+        Time Complexity: O(n * m), where n = length of strings array,  
+        and m = length of each string in the array
+        Note -- I'm not sure. Could you comment on this, please? Thank you!

```

Comment:
Sure! I personally think you're more right here to count m but we're also accepting just O(n). This is because since words in English don't usually get too long (around 5 letters), any good sized list will likely dwarf the effect of each of the letters.

(For what it's worth, I also wouldn't count the call to `sorted` as free since it is O(m * log(m)), but it seems we're not worrying about that. It'll be small too for similar reasons.)